Add a note to README about zlib?

Would a note about installing zlib libs to the README be appropriate? (e.g. `sudo apt-get install zlib1g-dev` -- depending on os/distro ofc).

I'm sort of surprised to even see this error. I've never run into it before. Is something weird on my new system? I don't ever remember installing zlib manually before. It's probably included in something I haven't installed yet.

On a relatively fresh Xubuntu 18.04 install, running `cargo install cargo-tree` fails with:

```
error: failed to run custom build command for `libssh2-sys v0.2.8`
process didn't exit successfully: `/tmp/cargo-installxLEdqE/release/build/libssh2-sys-b898ed33219b6dd9/build-script-build` (exit code: 101)
--- stdout
running: "cmake" "/home/nick/.cargo/registry/src/github.com-1ecc6299db9ec823/libssh2-sys-0.2.8/libssh2" "-DCRYPTO_BACKEND=OpenSSL" "-DBUILD_SHARED_LIBS=OFF" "-DENABLE_ZLIB_COMPRESSION=ON" "-DCMAKE_INSTALL_LIBDIR=lib" "-DBUILD_EXAMPLES=OFF" "-DBUILD_TESTING=OFF" "-DCMAKE_INSTALL_PREFIX=/tmp/cargo-installxLEdqE/release/build/libssh2-sys-3583025626f11749/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_BUILD_TYPE=Release"
-- The C compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.0g") 
-- Looking for EVP_aes_128_ctr
-- Looking for EVP_aes_128_ctr - found
-- Configuring incomplete, errors occurred!
See also "/tmp/cargo-installxLEdqE/release/build/libssh2-sys-3583025626f11749/out/build/CMakeFiles/CMakeOutput.log".

--- stderr
fatal: not a git repository (or any of the parent directories): .git
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.10/Modules/FindZLIB.cmake:112 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  src/CMakeLists.txt:239 (find_package)


thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', /home/nick/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.31/src/lib.rs:643:5
stack backtrace:
   0:     0x55ddcc46247e - std::sys::unix::backtrace::tracing::imp::unwind_backtrace::hd95e7c7781a77231
                               at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1:     0x55ddcc46af56 - std::sys_common::backtrace::print::h5c2c751288aab677
                               at libstd/sys_common/backtrace.rs:71
                               at libstd/sys_common/backtrace.rs:59
   2:     0x55ddcc45248b - std::panicking::default_hook::{{closure}}::h613226a9bb10326e
                               at libstd/panicking.rs:211
   3:     0x55ddcc452163 - std::panicking::default_hook::h8250355a3b9c55df
                               at libstd/panicking.rs:227
   4:     0x55ddcc452a5e - std::panicking::rust_panic_with_hook::h6617a62274b53c0e
                               at libstd/panicking.rs:463
   5:     0x55ddcc4525fc - std::panicking::begin_panic_fmt::hc7dc6580ccb0884f
                               at libstd/panicking.rs:350
   6:     0x55ddcc42eaa1 - cmake::fail::h6420f7e79e64c5ef
   7:     0x55ddcc42e64a - cmake::run::h82058d9c0c09139e
   8:     0x55ddcc42c469 - cmake::Config::build::h5f358fc9b7b9dae2
   9:     0x55ddcc42575b - build_script_build::main::hc20fa1eadd93dc78
  10:     0x55ddcc426892 - std::rt::lang_start::{{closure}}::h489a13ae2c59af47
  11:     0x55ddcc452542 - std::panicking::try::do_call::h64129d2b0e54f3b8
                               at libstd/rt.rs:59
                               at libstd/panicking.rs:310
  12:     0x55ddcc471fd9 - __rust_maybe_catch_panic
                               at libpanic_unwind/lib.rs:105
  13:     0x55ddcc45af25 - std::rt::lang_start_internal::h6264a91317866dd6
                               at libstd/panicking.rs:289
                               at libstd/panic.rs:374
                               at libstd/rt.rs:58
  14:     0x55ddcc426883 - main
  15:     0x7f90f43c2b96 - __libc_start_main
  16:     0x55ddcc423ac9 - _start
  17:                0x0 - <unknown>

warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-tree v0.19.0`, intermediate artifacts can be found at `/tmp/cargo-installxLEdqE`

Caused by:
  build failed

```

`sudo apt-get install zlib1g-dev` fixes.

--- a/README.md
+++ b/README.md
@@ -7,6 +7,12 @@
 
 Rust bindings to libssh2
 
+## Usage
+
+Ensure that [`zlib`](https://zlib.net/) is installed (Debian: `sudo apt-get
+install zlib1g-dev`, OSX: `brew install zlib-devel`,
+[Windows](http://gnuwin32.sourceforge.net/packages/zlib.htm))
+
 ```toml
 # Cargo.toml
 [dependencies]

